src/core/searcher.rs

```diff
@@ -77,11 +77,17 @@ impl Searcher {
         index: Index,
         segment_readers: Vec<SegmentReader>,
         generation: TrackedObject<SearcherGeneration>,
+        doc_store_cache_size: usize,
     ) -> io::Result<Searcher> {
         let store_readers: Vec<StoreReader> = segment_readers
             .iter()
             .map(SegmentReader::get_store_reader)
             .collect::<io::Result<Vec<_>>>()?;
+
+        for store_reader in &store_readers {
+            store_reader.set_cache_size(doc_store_cache_size);
```

Comment:
why could not this be done at the SegmentReader creation?

Reply:
yes I changed it